code.rs

```diff
@@ -164,7 +165,11 @@ pub fn main() {
         let rt = Builder::new_current_thread().enable_all().build().unwrap();
 
         let key_pair = KeyPair::new();
-        let config = Configuration::default();
+        let file_sqlite = gio::File::for_uri(&workbench::resolve("./db.sqlite"));
+        let config = Configuration {
+            database_url: format!("sqlite://{}", file_sqlite.path().unwrap().display()),
```

Comment:
`.display()` seems wrong

Reply:
```suggestion
            database_url: format!("sqlite://{}", file_sqlite.path().unwrap().to_str().unwrap()),
```

Slightly more correct. Paths are already annoying in Rust, since Rust doesn't assume that paths are valid UTF-8 strings and therefore checks a lot. The combination with `gio::File` makes that even worse.

Reply:
Should I use something else than gio::File ?

workbench::resolve returns a `file:///home/sonny/....` URI

We just need a `sqlite:///home/sonny/...` URI

Reply:
I think the way it currently is, is the way to go.